test/e2e/cluster_samples_operator_test.go

```diff
@@ -53,4 +91,157 @@ func TestClusterSamplesOperator(t *testing.T) {
 	if err != nil {
 		t.Errorf("error waiting for example imagestreamtag to appear: %v", err)
 	}
+
+	is := &imageapiv1.ImageStream{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ImageStream",
+			APIVersion: imageapiv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "jenkins",
+			Namespace: "openshift",
+		},
+	}
+
+	err = wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
+		if err := sdk.Get(is); err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Errorf("error waiting for example imagestream to appear: %v", err)
+	}
+
+}
+
+func verifyTemplatesPresent(t *testing.T) {
+	// as the content will vary over time as openshift/library evolves
+	// sample a random set of templates related artifacts
+	templates := []string{"jenkins-ephemeral", "redis-ephemeral"}
+	template := &templatev1.Template{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Template",
+			APIVersion: templatev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "openshift",
+		},
+	}
+
+	for _, name := range templates {
+		template.Name = name
+		err := wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
+			if err := sdk.Get(template); err != nil {
+				return false, nil
+			}
+			return true, nil
+		})
+		if err != nil {
+			t.Errorf("error waiting for example template %s to appear: %v", name, err)
+		}
+	}
+}
+
+func TestImageStreamInOpenshiftNamespace(t *testing.T) {
+	verifyOperatorUp(t)
+	verifyImagesPresent(t)
+	verifyTemplatesPresent(t)
+}
+
+func TestRecreateSamplesResourceAfterDelete(t *testing.T) {
+	sr := verifyOperatorUp(t)
+
+	oldTime := sr.CreationTimestamp
+
+	err := sdk.Delete(sr, sdk.WithDeleteOptions(&metav1.DeleteOptions{}))
+	if err != nil {
+		t.Errorf("error deleting samplesresource %v", err)
+	}
+
+	err = wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
+		if err := sdk.Get(sr); err != nil {
+			return false, nil
+		}
+		if sr.CreationTimestamp == oldTime {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Errorf("creation times the same after delete: %v, %v, %#v", oldTime, sr.CreationTimestamp, sr)
+	}
+
+	err = verifyConditionsCompleteSamplesAdded(sr)
+	if err != nil {
+		t.Errorf("samples not re-established after delete %#v", sr)
+	}
+
+	verifyImagesPresent(t)
+	verifyTemplatesPresent(t)
+}
+
+func TestManagedState(t *testing.T) {
```

Comment:
this is testing managed+removed states for the management state, so either rename appropriately(TestManagedAndRemovedState) or add testing for unmanaged state as well (and rename it to TestManagementState since you're testing the whole field)

Reply:
In my mind we were testing the `managementField` and varions valid values for it

Considering `TestSpecManagementStateField` given the prior name was unclear

And yes as I indicated in the description these tests in general are incomplete ... the needed delete processing changes I noted, which as I suspected has you rethinking the algorithm, resulted in my creating the PR now